test/mitmproxy/net/test_tcp.py

```diff
@@ -387,6 +387,13 @@ def test_echo(self):
             assert c.sni == "foo.com"
             assert c.rfile.readline() == b"foo.com"
 
+    def test_idn(self):
+        c = tcp.TCPClient(("127.0.0.1", self.port))
+        with c.connect():
+            c.convert_to_ssl(sni="кц.рф")
+            assert c.sni.encode("idna").decode("ascii") == "xn--j1ay.xn--p1ai"
```

Comment:
Yes it would. This is what I changed in `convert_to_ssl`  which is why I thought of writing it
Should I remove it?